docs/sources/developers/http_api/datasource_permissions.md

```diff
@@ -27,84 +27,137 @@ title: Datasource Permissions HTTP API
+| Action                       | Scope                                                                                      |
+| ---------------------------- | ------------------------------------------------------------------------------------------ |
+| datasources.permissions:read | datasources:\*<br>datasources:uid:\*<br>datasources:uid:my_datasource (single data source) |
 
 ### Examples
 
 **Example request:**
 
 ```http
-POST /api/datasources/1/enable-permissions
+GET /api/datasources/my_datasource/permissions HTTP/1.1
 Accept: application/json
 Content-Type: application/json
 Authorization: Bearer eyJrIjoiT0tTcG1pUlY2RnVKZTFVaDFsNFZXdE9ZWmNrMkZYbk
-
-{}
 ```
 
 **Example response:**
 
 ```http
 HTTP/1.1 200 OK
 Content-Type: application/json; charset=UTF-8
-Content-Length: 35
+Content-Length: 551
 
-{"message":"Datasource permissions enabled"}
+[
+    {
+        "id": 1,
+        "roleName": "fixed:datasources:reader",
+        "isManaged": false,
+        "isInherited": false,
+        "isServiceAccount": false,
+        "userId": 1,
+        "userLogin": "admin_user",
+        "userAvatarUrl": "/avatar/admin_user",
+        "actions": [
+            "datasources:read",
+            "datasources:query",
+            "datasources:read",
+            "datasources:query",
+            "datasources:write",
+            "datasources:delete"
+        ],
+        "permission": "Edit"
+    },
+    {
+        "id": 2,
+        "roleName": "managed:teams:1:permissions",
+        "isManaged": true,
+        "isInherited": false,
+        "isServiceAccount": false,
+        "team": "A team",
+        "teamId": 1,
+        "teamAvatarUrl": "/avatar/523d70c8551046f441727d690431858c",
+        "actions": [
+            "datasources:read",
+            "datasources:query"
+        ],
+        "permission": "Query"
+    },
+    {
+        "id": 3,
+        "roleName": "basic:admin",
+        "isManaged": false,
+        "isInherited": false,
+        "isServiceAccount": false,
+        "builtInRole": "Admin",
+        "actions": [
+            "datasources:query",
+            "datasources:read",
+            "datasources:write",
+            "datasources:delete"
+        ],
+        "permission": "Edit"
+    },
+]
 ```
 
 Status codes:
 
 - **200** - Ok
-- **400** - Permissions cannot be enabled, see response body for details
 - **401** - Unauthorized
 - **403** - Access denied
-- **404** - Datasource not found
+- **500** - Internal error
+
+## Add or revoke access to a data source for a user
 
-## Disable permissions for a data source
+`POST /api/access-control/datasources/:uid/users/:id`
 
-`POST /api/datasources/:id/disable-permissions`
+Sets user permission for the data source with the given `uid`.
 
-Disables permissions for the data source with the given `id`. All existing permissions will be removed and anyone will be able to query the data source.
+To add a permission, set the `permission` field to either `Query` or `Edit`.
```

Comment:
Didn't you add `Admin` lately?

Reply:
Well spotted! I had this PR open beforehands and forgot to update it.